### PR TITLE
[Android] Remove the ". " on empty labels (Accessibility) on Fastrenderers

### PR DESCRIPTION
**Pull request by samhouts**
### Description of Change ###

This is an alternative to #904 (and revision to #801). This will clarify and simplify the code a bit more. 

Note that this code will be moved to extension methods if #889 is accepted.

Also adds a UI test to prevent recurrence.

Also removed the nullable parameters in the `AutomationPropertiesProvider` that are no longer necessary.

### Bugs Fixed ###

- Reported in #904 and #801
- [Bug 56444 - A small "dot" shows up as value in Android labels when the value is null/empty.](https://bugzilla.xamarin.com/show_bug.cgi?id=56444)

### API Changes ###

None

### Behavioral Changes ###

None

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/FastRenderers/AutomationPropertiesProvider.cs [hunk 1]
```diff
@@ -142,18 +132,27 @@ bool SetHint(string hint = GetFromElement)
...
+			string separator;
+			var name = (string)Element.GetValue(AutomationProperties.NameProperty);
+			var helpText = (string)Element.GetValue(AutomationProperties.HelpTextProperty);
+
+			if (string.IsNullOrWhiteSpace(name) || string.IsNullOrWhiteSpace(helpText))
+				separator = "";
+			else
+				separator = ". ";
+
+			return $"{name}{separator}{helpText}";
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.Android/FastRenderers/AutomationPropertiesProvider.cs]: 1/ Are you sure you mean `IsNullOrWhiteSpace` and not `IsNullOrEmpty` ?

2/ Should be:
```
if (string.IsNullOrWhiteSpace(name)
    return helpText;
if (string.IsNullOrWhiteSpace(helpText)
    return name;

return $"{name}. {helpText}"
```
so you don't concatenate empty strings (this 2/ is only valid if you don't care about whitespaces

File: Xamarin.Forms.Platform.Android/FastRenderers/AutomationPropertiesProvider.cs [hunk 2]
```diff
@@ -142,18 +132,27 @@ bool SetHint(string hint = GetFromElement)
...
-					(string)Element.GetValue(AutomationProperties.HelpTextProperty));
-			}
+			string value = ConcatenateNameAndHelpText(Element);
 
 			textView.Hint = !string.IsNullOrWhiteSpace(value) ? value : _defaultHint;
 
 			return true;
 		}
 
+		static string ConcatenateNameAndHelpText(Element Element)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.Android/FastRenderers/AutomationPropertiesProvider.cs]: I'd AggressiveInline this

File: Xamarin.Forms.Platform.Android/VisualElementRenderer.cs [hunk 3]
```diff
@@ -411,6 +411,20 @@ protected virtual bool SetHint()
 			return true;
 		}
 
+		static string ConcatenateNameAndHelpText(Element Element)
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Platform.Android/VisualElementRenderer.cs]: do we need this declared twice ?

**Comment by BradChase2011** [comment]
@samhouts Saam, do you know how far away this is from being accepted?  Merge wise, do we need to merge this with other PRs or are we good just merging this in for the time being until its accepted?

**Pull request merged**